staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -251,6 +252,15 @@ type Interface interface {
 	// TODO: Remove when storage.Interface will be separate from etc3.store.
 	// Deprecated: Added temporarily to simplify exposing RequestProgress for watch cache.
 	RequestWatchProgress(ctx context.Context) error
+
+	// Supports checks whether the etcd instance supports the specified features and returns true if it does,
+	// or false otherwise. It utilizes the provided context.Context for cancellation or timeout control.
+	// The function accepts one or more featuregate.Feature parameters representing the features to check.
+	//
+	// TODO(ah8ad3): Consider enhancing the feature checking mechanism by potentially combining
+	// featuregate.Feature with additional feature lists. Implementing a pattern like Visitor may offer
+	// improvements in extensibility and maintainability.
+	Supports(ctx context.Context, features ...featuregate.Feature) (bool, error)
```

Comment:
As @jpbetz points out, this seems better.
Also i've added `FeatureSupportChecker` in `store.go` to make it easier for future to add more checkers if it's convenient for us. 
I know it may not be the right PR to add `FeatureSupportChecker` or it may not be useful in future but anyway if we don't want it i can remove it np. Also you may be interested too @p0lyn0mial maybe it can help (base on this https://github.com/kubernetes/kubernetes/pull/123264#discussion_r1495410968).
Please take a look. @jpbetz @wojtek-t @serathius @sftim .
Thanks for you're time.

Reply:
I like it.

That said, let's swtich it to just take a single feature as a parameter - variadic paramters are discouraged and I don't see a real need take it [a caller may always call it for different things they may want to check]

Reply:
What I'm missing is - how exactly you're going to use this to achieve what we want to achieve (disable consistent lists)? Can you add that in a separate commit?